chain-impl-mockchain/src/leadership/mod.rs

```diff
@@ -204,12 +201,6 @@ impl Leadership {
         &self.inner
     }
 
-    /// access the ledger parameter for the current leadership
-    #[inline]
-    pub fn ledger_parameters(&self) -> &LedgerParameters {
```

Comment:
I think this could be useful to calculate in advance the leadership schedule without having an actual ledger available, have you thought about this?

Reply:
Good point, need to think how it is better organize, taking a look...

Reply:
Updated, I have modified `LeadershipData` generation, but `Leadership` remain the same, if we will pass all this arguments separately will have `clippy:: too_many_arguments` warning.